[CI] Add NuGet MSI conversion and VS insert stage

pjcollins opened the pull request:
Context: https://github.com/xamarin/yaml-templates/pull/117
Context: https://github.com/xamarin/yaml-templates/pull/121

Updates `BundledVersions.targets` and `WorkloadManifest.json` exclude
package semver metadata, as this is excluded in both Android and maciOS.
The .msi conversion tooling also does not process these semver metadata
correctly.

A new CI job and stage have been added to facilitate the Visual Studio
installer insertion process.

The first job named "Convert NuGet to MSI" will convert signed packages
to .msi installers, and generate Visual Studio manifests for the .msi
installers.  The .msi files are signed and converted back to .nupkg
files for the Windows command line install scenario.  These packages are
uploaded as pipeline artifacts named `vsdrop-signed`, and 
`vs-msi-nugets`.

The new `vs-workload.props` file contains version information and other
metadata used by the convert job to generate a Visual Studio manifest.

The new "VS Insertion" stage starts with a [manual validation task][0].
This task will pause execution and wait for someone to click a "Resume"
or "Reject" button that will appear on the pipeline UI.  This task is
configured to be rejected after waiting for two days, but it can be
manually re-ran any time after being rejected to trigger a VS insertion
for an older build.

If the manual validation task is approved, a VS Drop will be created
containing all .NET 6 .msi files.  This Drop URL can then be used to
update our component versions in Visual Studio.  This last piece is
currently manual as we will initially be introducing new components,
however we should be able to automate VS PR creation in the future.

[0]: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/manual-validation?view=azure-devops&tabs=yaml

pjcollins committed:
[WIP] Add nuget convert and VS insert jobs

pjcollins committed:
Fix xml tag

pjcollins committed:
Use staged yaml-templates branch

pjcollins committed:
Revert stage name changes, remove metadata from framework ref

pjcollins committed:
Set TeamName globally

pjcollins committed:
Fix signed .nupkg wildcard

pjcollins committed:
Specify nested signed .nupkg path

pjcollins committed:
Update component version

pjcollins committed:
Merge remote-tracking branch 'origin/main' into pjc/nuget-convert-insert

pjcollins committed:
Rename workload names and .props

pjcollins committed:
Shorten AspNetCore WebView name

pjcollins committed:
Update to new Android/maciOS workload names

pjcollins committed:
Build fixes, use maccatalyst rid in output paths

pjcollins committed:
Merge remote-tracking branch 'origin/main' into pjc/nuget-convert-insert

pjcollins committed:
Revert empty packs addition after bringing in arcade fix

pjcollins committed:
Shorten more wl pack names

pjcollins committed:
Sign missing files

rmarinho commented:
Do we want to merge to yaml-templates before taking this ?

jonathanpeppers commented on eng/automation/vs-workload.template.props:
```diff
@@ -0,0 +1,56 @@
+    </ShortNames>
+
+    <ShortNames Include="microsoft.maui.essentials.runtime;Microsoft.Maui.Essentials.Runtime">
+      <Replacement>MMEss.RT</Replacement>
```
How short do these have to be? Would a name like `Essentials.Runtime` still be too long?

pjcollins replied:
We have a 64 character limit from what I've seen in my testing, but this includes the full version string that we add to file names. I am currently looking at trimming the version information instead this name string, as the version information will still be associated with the pack even if it is not in the file name.

pjcollins replied:
We now replace the semver version in the pack names with the shorter four part version that is used for the generated VS components. This has a more consistent length and should be all that is needed to avoid path issues during the VS build.

jonathanpeppers commented:
> Do we want to merge to yaml-templates before taking this ?

Yes, I think we need this first: https://github.com/xamarin/yaml-templates/pull/117

It looks like there are some temporary changes to bump iOS & Android here, too.

pjcollins committed:
Update names, short names, descriptions

pjcollins committed:
Remove extensions from skip list

pjcollins committed:
Don't skip System.CodeDom.dll signing

pjcollins committed:
Merge remote-tracking branch 'origin/main' into pjc/nuget-convert-insert

 Conflicts:
	eng/Version.Details.xml
	eng/Versions.props
	src/Workload/Microsoft.Maui.Controls.Ref/Microsoft.Maui.Controls.Ref.csproj
	src/Workload/Microsoft.NET.Sdk.Maui/WorkloadManifest.in.json

pjcollins committed:
Merge cleanup for maciOS workload version update

pjcollins commented:
Latest test run triggered here https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4998323&view=results.

pjcollins commented:
@mattleibow @rmarinho when you have a chance to review, I think we can land this for now and then take any future incremental changes as needed for preview 7.

mattleibow committed:
Merge remote-tracking branch 'origin/main' into pjc/nuget-convert-insert

pjcollins commented:
Let's hold off on this for now, we have more yaml-templates changes coming in hot to handle .msi generation for manifest packs.

jonathanpeppers reviewed (approved):
LGTM, but I think @pjcollins wanted to get one more change from a .yaml template before merging this.

dotnet-maestro[bot] committed:
[main] Update dependencies from xamarin/xamarin-android (#1767)

* Update dependencies from https://github.com/xamarin/xamarin-android build 6.0.1xx-preview7-8ca289588b6969a2cc878afa0813afec8cd63dd3-1

Microsoft.Android.Sdk.Windows
 From Version 30.0.100-preview.7.91 -> To Version 30.0.100-preview.7.93

Dependency coherency updates

Microsoft.NETCore.App.Ref,Microsoft.Dotnet.Sdk.Internal,Microsoft.Extensions.Primitives,Microsoft.AspNetCore.App.Runtime.win-x64,Microsoft.Extensions.Hosting.Abstractions,Microsoft.Extensions.DependencyInjection.Abstractions,Microsoft.Extensions.DependencyInjection,Microsoft.Extensions.Configuration.Abstractions,Microsoft.Extensions.Configuration,Microsoft.Extensions.Logging.Abstractions,Microsoft.Extensions.Logging,Microsoft.Extensions.Logging.Console,Microsoft.Extensions.Logging.Debug,Microsoft.Extensions.FileProviders.Abstractions,Microsoft.Extensions.FileProviders.Embedded,Microsoft.AspNetCore.Authorization,Microsoft.AspNetCore.Components.WebView,Microsoft.AspNetCore.Components.Web,Microsoft.JSInterop,Microsoft.WindowsDesktop.App.Runtime.win-x64,System.CodeDom
 From Version 6.0.0-preview.7.21326.8 -> To Version 6.0.0-preview.7.21368.2 (parent: Microsoft.Dotnet.Sdk.Internal

* [build] remove workaround for iOS AOT packages on Windows

* Ignore NU1703 in DeviceTests

NU1703 warns about using nuget packages which have xamarin.ios tfm references from a newer net6.0-maccatalyst project.

* Workaround error for implicit imports

Context: https://github.com/dotnet/sdk/issues/19050
and https://github.com/xamarin/java.interop/commit/4fb7c147f8c6eb9bf94d9bfb8305c7d2a7a9fb33

* Ignore NU1703 for Mac Catalyst projects

* Revert "Ignore NU1703 in DeviceTests"

This reverts commit 8fdf5e10a292d824fb2c99e7b32bbbe8c81fb4e2.

* Revert "Revert "Ignore NU1703 in DeviceTests""

This reverts commit 4d0456d558b4be5e41a6ee8a6217641162a34c6d.

* Fix other C# 10-related errors

* Use `LangVersion=latest` by default
* Fixed new NRT-related errors that appeared from C# 10.0

* Disable StaticWebAssets in MauiRazorClassLibrarySample

Workaround to get the build to work. Will revert later to investigate.

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Jonathan Peppers <jonathan.peppers@microsoft.com>
Co-authored-by: Jonathan Dick <jondick@gmail.com>
Co-authored-by: Eilon Lipton <Eilon@users.noreply.github.com>

hartez committed:
IContainer as IList<IView> (#1724)

* IContainer proposal

* Update the test classes to match the updated interfaces

* Make substitution work for new layout interface

* Fix Windows

* Implement methods on stub I missed

* And another thing I forgot to implement

* Remove commented code

* Make IContainer implementation explicit

* Fix child collection access in test

PureWeen committed:
Shell Android Handler (#979)

* Android Shell Handler

* - fix resource references on compatibility

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/xamarin/xamarin-android build 6.0.1xx-preview7-1734a83c4788337258aa4e83d45a6f143500b7cb-1 (#1783)

Microsoft.Android.Sdk.Windows
 From Version 30.0.100-preview.7.93 -> To Version 30.0.100-preview.7.95

Dependency coherency updates

Microsoft.Extensions.Hosting.Abstractions,Microsoft.Extensions.DependencyInjection.Abstractions,Microsoft.Extensions.DependencyInjection,Microsoft.Extensions.Configuration.Abstractions,Microsoft.Extensions.Configuration,Microsoft.Extensions.Logging.Abstractions,Microsoft.Extensions.Logging,Microsoft.Extensions.Logging.Console,Microsoft.Extensions.Logging.Debug,Microsoft.Extensions.FileProviders.Abstractions,System.CodeDom
 From Version 6.0.0-preview.7.21365.2 -> To Version 6.0.0-preview.7.21368.2 (parent: Microsoft.AspNetCore.App.Runtime.win-x64

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

pjcollins committed:
Update to use msi conversion v2

pjcollins committed:
Merge remote-tracking branch 'origin/main' into pjc/nuget-convert-insert

pjcollins commented:
Test run for v2 msi conversion: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5013484&view=results

pjcollins committed:
Use NuGetPackagePath for .nupkg dir

pjcollins committed:
Explicitly define System assemblies to skip signing for

pjcollins committed:
Use main template branch

pjcollins committed:
Merge remote-tracking branch 'origin/main' into pjc/nuget-convert-insert
